### PR TITLE
Bf16 fix

### PR DESCRIPTION
**Pull request by josephdviviano**
- [x] I've read the [.github/CONTRIBUTING.md](.github/CONTRIBUTING.md) file
- [x] My code follows the typing guidelines
- [x] I've added appropriate tests
- [x] I've run pre-commit hooks locally

## Description
small fix to prevent error observed in https://github.com/GFNOrg/torchgfn/pull/367

### CODE REVIEW

File: src/gfn/estimators.py [hunk 1]
```diff
@@ -304,7 +307,13 @@ def to_probability_distribution(
...
             logits /= temperature
 
-        probs = torch.softmax(logits, dim=-1)
+        # Disable autocast (always use fp32) to avoid numerical instability issues.
+        with torch.autocast(
+            device_type="cuda" if logits.is_cuda else "cpu", enabled=False
+        ):
+            probs = torch.softmax(logits, dim=-1)
+            probs = probs + numerical_stability_eps  # Add small epsilon for stability.
+            probs = probs / probs.sum(dim=-1, keepdim=True)
```
[the remark this answers is not in the document]
Comment (josephdviviano) [reply src/gfn/estimators.py:316]: this is a numerical stability trick 

"Small Epsilon for Logarithms: In other contexts, a small positive value \(\epsilon \) might be added to the softmax output before taking a logarithm (e.g., in cross-entropy loss) to prevent issues with \(\log (0)\) if a softmax output becomes exactly zero due to underflow. This is a common practice in numerical computations involving probabilities that can theoretically reach zero."

I'm not entirely sure if this was the source of the fix, but it seems like a fine thing to leave in.

[the remark this answers is not in the document]
Comment (hyeok9855) [reply src/gfn/estimators.py:316]: I guess we need to use probs to apply epsilon-exploration.

[the remark this answers is not in the document]
Comment (younik) [reply src/gfn/estimators.py:316]: > We could also use logits for everything and the cast to probabilities just for epsilon exploration. What do you think?

I think this would be ideal, but of course require more work. If this fixes the problem, I am fine merging it.

I am wondering if we can do epsilon exploration directly on logits

[the remark this answers is not in the document]
Comment (younik) [reply src/gfn/estimators.py:316]: > We could also use logits for everything and the cast to probabilities just for epsilon exploration. What do you think?

I think this would be ideal, but of course require more work. If this fixes the problem, I am fine merging it.
